espnet2/train/trainer.py

```diff
@@ -127,15 +162,11 @@ def train_one_epoch(
                 loss, stats, weight = data_parallel(
                     model, (), range(ngpu), module_kwargs=batch
```

Comment:
Some developers in pytorch and fairseq says that DataParallel, 1-Node-n-GPU with multithread, is not efficient and use DistributedDataParallel, m-Node-mn-GPU with multiprocess.
Multithread has disadvantage due to Global-interpreter-lock of Python interpreter in general.
We might remove the supporting DataParallel and unify it to DistributedDataParallel.